datamanagement/testing/tantalus_filepath_check.py

```diff
@@ -0,0 +1,88 @@
+import os
+import sys
+from dbclients.tantalus import TantalusApi
+
+
+def init():
+    print "STARTING, NOT DELETING"
+
+    tantalus_api = TantalusApi()
+    tantalus_filecheck_result = open("tantalus_filepath_check_result.txt", "w+")
+    fail_flag = False
+
+    print "Collecting File Resources..."
+
+    file_resources = tantalus_api.list('file_resource', fileinstance__storage__name="shahlab")
+
+    print "File Resources collected..."
+
+    assert file_resources, "Empty File Resources is empty"
+
+    shahlabclient = tantalus_api.get_storage_client("shahlab")
+    singlecellblob = tantalus_api.get_storage_client("singlecellblob")
+
+    print "Blob Storage collected..."
+
+    for file_resource in file_resources:
+        file_instances = {}
+        date_flag = False
+        for file_instance in file_resource["file_instances"]:
+            file_instances[file_instance['storage']['name']] = file_instance
+
+        print "Checking " + file_instances['shahlab']['filepath']
+        if shahlabclient.exists(file_instances['shahlab']['filepath']):
+            if  shahlabclient.get_size(file_instances['shahlab']["filepath"]) != file_resource["size"] :
+                print "Expected size do not match the actual size!"
+
+                tantalus_filecheck_result.write(
+                    "\n ERROR: Size mismatch in: " + file_instances['shahlab']['filepath'] +
+                    "\nActual Size: " + str(shahlabclient.get_size(file_instances['shahlab']['filepath'])) +
+                    "\nExpected Size: " + str(file_resource["size"]) + "\n"
+                )
+
+                tantalus_filecheck_result.write("Tantalus: ")
+                tantalus_filecheck_result.write(file_resource["created"])
+                tantalus_filecheck_result.write("\nShahlab: ")
+                tantalus_filecheck_result.write(shahlabclient.get_created_time(file_instances['shahlab']['filepath']))
+
+                if 'singlecellblob' in file_instances.keys():
+                    file_instances['singlecellblob']['filepath'] =  file_instances['singlecellblob']['filepath'].replace("singlecelldata/data/","")
+                    if singlecellblob.exists(file_instances['singlecellblob']['filepath']):
+                        tantalus_filecheck_result.write("\nSinglecellBlob: ")
+                        tantalus_filecheck_result.write(singlecellblob.get_created_time(file_instances['singlecellblob']['filepath']))
+                        if singlecellblob.get_created_time(file_instances['singlecellblob']['filepath']) == file_resource["created"]:
+                            date_flag = True
+
+                if file_resource["created"] > shahlabclient.get_created_time(file_instances['shahlab']['filepath']) or date_flag:
```

Comment:
This should be and. Specifically, we only delete if the shahlab version is older than than the date in tantalus, _and_ the blob version date is the same as that in tantalus.